btree.go

```diff
@@ -64,6 +64,42 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+}
+
+func NewFreeList() *FreeList {
+	return &FreeList{freelist: make([]*node, 0, 32)}
+}
+
+func (f *FreeList) newNode(t *BTree) (n *node) {
+	index := len(f.freelist) - 1
+	if index < 0 {
+		return &node{t: t}
+	}
+	f.freelist, n = f.freelist[:index], f.freelist[index]
+	n.t = t
```

Comment:
I'd rather any modification of the node be done in the BTree code, not the freelist code.  IE: line 84 here and lines 90-98 in freeNode.  Let's keep FreeList as simple as possible, have it hold nodes but not do anything with them.